huaweicloud/services/dns/resource_huaweicloud_dns_line_group.go

```diff
@@ -0,0 +1,369 @@
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	createDNSLineGroupClient, err := cfg.NewServiceClient("dns", region)
+	if err != nil {
+		return diag.Errorf("error creating DNS client: %s", err)
+	}
+
+	// createDNSLineGroup: Create the DNS line group
+	if err := createDNSLineGroup(createDNSLineGroupClient, d); err != nil {
+		return diag.FromErr(err)
+	}
+
+	timeout := d.Timeout(schema.TimeoutCreate)
+	if err := waitForDNSLineGroupCreateOrUpdate(ctx, createDNSLineGroupClient, d, timeout); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceDNSLineGroupRead(ctx, d, meta)
+}
+
+func createDNSLineGroup(lineGroupClient *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	var createDNSLineGroupHttpUrl = "v2.1/linegroups"
+
+	createDNSLineGroupPath := lineGroupClient.Endpoint + createDNSLineGroupHttpUrl
+	createDNSLineGroupOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			202,
+		},
+	}
+	createDNSLineGroupOpt.JSONBody = utils.RemoveNil(buildCreateOrUpdateDNSLineGroupBodyParams(d))
+	createDNSLineGroupResp, err := lineGroupClient.Request("POST", createDNSLineGroupPath,
+		&createDNSLineGroupOpt)
+	if err != nil {
+		return fmt.Errorf("error creating DNS line group: %s", err)
+	}
+
+	createDNSLineGroupRespBody, err := utils.FlattenResponse(createDNSLineGroupResp)
+	if err != nil {
+		return err
+	}
+
+	id, err := jmespath.Search("line_id", createDNSLineGroupRespBody)
+	if err != nil {
+		return fmt.Errorf("error creating DNS line group: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+	return nil
+}
+
+func waitForDNSLineGroupCreateOrUpdate(ctx context.Context, lineGroupClient *golangsdk.ServiceClient,
+	d *schema.ResourceData, timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Target:       []string{"ACTIVE"},
+		Pending:      []string{"PENDING_CREATE", "PENDING_UPDATE"},
+		Refresh:      dnsLineGroupStatusRefreshFunc(d, lineGroupClient),
+		Timeout:      timeout,
+		Delay:        5 * time.Second,
+		PollInterval: 5 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for DNS line group (%s) to be ACTIVE: %s", d.Id(), err)
+	}
+	return nil
+}
+
+func buildCreateOrUpdateDNSLineGroupBodyParams(d *schema.ResourceData) map[string]interface{} {
+	dnsLines := d.Get("lines").(*schema.Set).List()
+
+	return map[string]interface{}{
+		"name":        utils.ValueIngoreEmpty(d.Get("name")),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
+		"lines":       utils.ValueIngoreEmpty(dnsLines),
+	}
+}
+
+func resourceDNSLineGroupRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// getDNSLineGroup: Query the DNS line group detail.
+	var (
+		getDNSLineGroupHttpUrl = "v2.1/linegroups"
```

Comment:
why not call the ShowLineGroup API which URI is `/v2.1/linegroups/{linegroup_id}`